client/src/app/AppTypes.res

```diff
@@ -505,7 +505,7 @@ module SavedSettings = {
 
       {
         firstVisitToDark: oldFirstVisitToDark || newFirstVisitToDark,
-        recordConsent: withDefault(None, field("recordConsent", optional(bool)), j),
+        recordConsent: withDefault(default.recordConsent, field("recordConsent", bool), j),
```

Comment:
This needs to still be able to read old values, so it needs to support reading both the old type and the new type. I think `optional(bool)` will safely read regular bools, but worth checking.

Should add a comment to go with it.

Reply:
Isn't this why we're using withDefault or wouldn't it be enough alone? 
optional(bool) didn't work.

Reply:
Why didnt `optional(bool)` work? Was there an error message?

The alternative is to try to parse it multiple times using different decoders. As I recall, there's a bs-json function that takes multiple decoders as an argument and tries them in turn.

Reply:
- The error message is: 
This has type:  `Json_decode_extended.decoder<bool>`
Somewhere wanted: `Json_decode_extended.decoder<option<bool>>`

- Changing recordConsent type back to `option<bool>` will bring us back to where we started.

- There are two functions `oneOf` and `either`, both of them raise the same error.
`oneOf(list{optional(bool), field("recordConsent", bool)}, j),`
`either(optional(bool), field("recordConsent", bool), j),`

Is there something I'm missing or misunderstood?

Reply:
`oneOf` is the right function. Both decoder arguments need to have the same type, so you need a decoder that returns a `bool`. Which means you need to write a decoder which handles the optional case and returns a default. There are example decoders is Json_decode_extended.